### ISSUE TITLE
Html <a href prefetch> option to start request when mouse over a href link and not yet clicked

### ISSUE THREAD

**Issue by zxcq544** [WICG/proposals#66]
## Introduction

Currently when we click on links on web pages it starts to load only after we click on it even though we could predict this intention
when mouse was over link. It takes human brain around 150-300 milliseconds(judging by statistics at https://humanbenchmark.com/tests/reactiontime) to send signal to click on mouse when something gets under cursor. We could use this time to load whole page and even render it in background when user puts his cursor over <a href prefetch> link and after click replace current render with the one we made in background. This would feel as if link loaded instantly even though it took 150 ms to do it in background but since human brain doesn't click instantly we can make it feel as if page load was actually instant.

## Use Cases (Recommended)

This could be used for all <a href prefetch > clicks inside same domain. It gives huge user experience boost. All <a href prefetch > clicks feel instant.

## Goals (Optional)

Increase user experience.

## Non-goals (Optional)

Simplify creation of <a href> links with prefetch feature. Currently if you want to prefetch <a href> links you need to use js server like Sveltekit which prerenders links on mouse hover and replaces pages on click which makes it look like page was loaded instantly on click.

## Proposed Solution

Allow "prefetch" option inside <a href> links so it looks like <a href="/link" prefetch>. When mouse cursor is over such link then it shoud make normal HTTP GET request as normal link would do on click then render the page from request in background and then after click on link it should replace current page with the page that was rendered in background.

## Examples (Recommended)

For example we are at github.com. Let's take a look at header. We have <a href='/issues'>. With new option we could turn it into <a href='/issues' prefetch> and then if we put cursor over that element we would instantly make HTTP GET request to get data and render page for /issues in background. For me it takes 370 ms to get data from /issues and it takes 270 ms for my brain to click on link when it's under my cursor(I checked it here https://humanbenchmark.com/tests/reactiontime).
So if the link had prefetch option on it - it would feel like it loaded in 100 ms instead of 370 ms. Those 270 ms could make a huge difference in how responsive the page feels. 

## Alternate Approaches (Optional)

Currently same thing is done using javascript servers like Sveltekit.

## Privacy & Security Considerations

Those are normal links with different mouse over and click behaviour.

## Let’s Discuss (Optional)

What to show when link is not fully loaded and rendered.

**Comment by domenic** [comment]
Hey! I think this is a very interesting area, and is in fact what I'm spending the majority of my day job on :).

The community's current best attempts at this are taking place over on https://github.com/WICG/nav-speculation . In general, a lot of the complexity comes in speccing out what one of these prefetches does---does it trigger the normal navigation path? Does it mess with cookies? How does it affect performance timing APIs? Etc.

The specific trigger mechanism you propose, of annotating every link and using hover as the trigger, is desktop-specific and not very flexible; e.g. it doesn't work well for cases that use APIs like `location.href` or `window.open()`. Our current thinking is outlined in https://github.com/WICG/nav-speculation/blob/main/triggers.md , and relies on a JSON manifest of "speculation rules". One such variant is "document rules" which allow you to use a CSS selector, so e.g. your page could use a `[data-prefetch]` selector to get a similar experience to what you propose. The browser is then in charge of determining *when* to prefetch; whether that's on hover on desktop, on finger down on mobile, in idle time on either, or never if the user is in battery-saver mode. We're also discussing other more imperative triggers in https://github.com/WICG/nav-speculation/issues/98.

I'll close this issue, since we're still in the incubation stage for this idea, but please feel free to get involved over there! Much of the tech is already launched in Chromium, albeit still not that easy to use or comprehensive.

**Comment by cwilso** [comment]
As @domenic said, this is a use case for  https://github.com/WICG/nav-speculation.  If this isn't appropriate, please re-open.
